### PR TITLE
pipeline: decorate builds and exit early if no changes

### PR DESCRIPTION
**Pull request by jlebon**
First, learn to read what the previous build was and then skip the rsync
out stage if there are no new builds.

Second, decorate build descriptions of new builds with their build ID or
otherwise, just say no changes. This makes it easy at a glance to find
out which jobs created new builds.

### CODE REVIEW

**Comment by jlebon** [comment]
Requires: #24

File: Jenkinsfile [hunk 1]
```diff
@@ -24,11 +26,27 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
...
             """)
         }
+
+        def newBuildID = utils.shwrap_capture("readlink builds/latest")
+        if (prevBuildID == newBuildID) {
+            currentBuild.result = 'SUCCESS'
+            currentBuild.description = "💤 (no new build)"
+            return
+        } else {
+            currentBuild.description = "⚡ ${newBuildID}"
```
Comment (dustymabe) [inline Jenkinsfile]: i'm guessing this shows up in the jenkins web interface somehow?

Comment (jlebon) [reply Jenkinsfile]: Yup, exactly! It's in the build description text that appear in the build history on the left margin of the page.

File: Jenkinsfile [hunk 2]
```diff
@@ -24,11 +26,27 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
...
+
         stage('Build') {
             utils.shwrap("""
             coreos-assembler build
             """)
         }
+
+        def newBuildID = utils.shwrap_capture("readlink builds/latest")
+        if (prevBuildID == newBuildID) {
+            currentBuild.result = 'SUCCESS'
```
[the remark this answers is not in the document]
Comment (jlebon) [reply Jenkinsfile]: Correct.

[the remark this answers is not in the document]
Comment (jlebon) [reply Jenkinsfile]: (Not setting the `result`, but doing the `return` just under).

**Comment by jlebon** [comment]
Rebased!

File: Jenkinsfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dustymabe) [reply Jenkinsfile]: ahh, missed the `return` but I figured you were doing this for a reason. Exiting early was something I wanted to do. Thanks!

File: Jenkinsfile [hunk 3]
```diff
@@ -24,11 +26,27 @@ podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultCon
             coreos-assembler fetch
             """)
         }
+
+        def prevBuildID = null
+        if (utils.shwrap_rc("test -f builds/latest")) {
```
Comment (jlebon) [inline Jenkinsfile]: I'll note that there's a `fileExists` step I tried to use here, but it sadly doesn't work correctly inside the Kubernetes container. So we have to resort to just `test -f`.

**Pull request merged**